### PR TITLE
feat: Improve CH SQL migrations to actually raise on exceptions

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

It _appears_ that the built in CH `migrations.RunSQL` command doesn't actually error out if the command fails. Because we all write fantastic SQL code, this of course never happens </s> but just in case we should probably make sure it does.

## Changes

* Swap out all uses of `migrations.RunSQL` for our own custom command that calls `sync_execute` instead. This might not be the best approach but I wanted to try it and see what if the CI catches anything funny.

## How did you test this code?

Ran the migrations myself locally and hoping the CI picks up anything else.

> NOTE: I would definitely like some feedback from experts here. This is more of a prompting PR than a ready-to-go one

### CODE REVIEW

File: posthog/clickhouse/client/migration_tools.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from infi.clickhouse_orm import migrations
+
+from posthog.clickhouse.client.execute import sync_execute
+
+
+def run_sql_with_exceptions(sql: str):
```
[the remark this answers is not in the document]
Comment (benjackwhite) [reply posthog/clickhouse/client/migration_tools.py:6]: Good point

[the remark this answers is not in the document]
Comment (hazzadous) [reply posthog/clickhouse/client/migration_tools.py]: FYI looks like the migration library we're using picks up all modules from the migrations path atm, and interprets the `migration_tools` as a migration, thereby failing the migration (see failing tests)

[the remark this answers is not in the document]
Comment (benjackwhite) [reply posthog/clickhouse/client/migration_tools.py:6]: Indeed thats why I didn't do it 😅

**Review by hazzadous — APPROVED** [review]
Looks good

I'm not sure however if the infi RunSQL e.g. sets some ClickHouse settings that would be useful to keep.

**Comment by hazzadous** [comment]
@benjackwhite need any help with this? Seems like an annoying source of potential instance killers!

**Comment by benjackwhite** [comment]
Yeah I wasn't sure if I need to do something based on your previous comment @hazzadous 

If you're happy, this can be merged whenever, just haven't had the time to keep on it 😅

**Comment by hazzadous** [comment]
> Yeah I wasn't sure if I need to do something based on your previous comment @hazzadous

Hah neither am I, what's the worst that could happen though?  😅  Chance of breaking deploys, possible. Chance of deleting data/tables, pretty low. I'd say send it.

**Pull request merged**